crypto/bio/bf_buff.c

```diff
@@ -380,6 +385,7 @@ static long buffer_ctrl(BIO *b, int cmd, long num, void *ptr)
             }
         }
         ret = BIO_ctrl(b->next_bio, cmd, num, ptr);
+        BIO_copy_next_retry(b);
```

Comment:
The blanket calls to `BIO_copy_next_retry()` for all possible IO ctrls does not seem correct to me. Most of those ctrls are not "IO" related ctrls so applications would not expect the "retry" the status to change as a result. The only one where it would is in BIO_CTRL_FLUSH. So I would keep the two you have inserted in this section and remove all the others.

Reply:
@mattcaswell 
Thanks! I have removed all other insertions. In the new pushed commit I also added some for other BIO_CTRL_FLUSH. Is it correct?

Reply:
ping~ @mattcaswell